channeldb/edge_info.go

```diff
@@ -1,35 +1,137 @@
 package channeldb
 
 import (
+	"bufio"
 	"bytes"
 	"encoding/binary"
 	"errors"
+	"fmt"
 	"io"
 
 	"github.com/btcsuite/btcd/wire"
 	"github.com/lightningnetwork/lnd/channeldb/models"
 	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/tlv"
 )
 
+// edgeInfoEncodingType indicate how the bytes for a channel edge have been
+// serialised.
+type edgeInfoEncodingType uint8
+
+const (
+	// edgeInfo2EncodingType will be used as a prefix for edge's advertised
+	// using the ChannelAnnouncement2 message. The type indicates how the
+	// bytes following should be deserialized.
+	edgeInfo2EncodingType edgeInfoEncodingType = 0
+)
+
+const (
+	// EdgeInfo2MsgType is the tlv type used within the serialisation of
+	// ChannelEdgeInfo2 for storing the serialisation of the associated
+	// lnwire.ChannelAnnouncement2 message.
+	EdgeInfo2MsgType = tlv.Type(0)
+
+	// EdgeInfo2Sig is the tlv type used within the serialisation of
+	// ChannelEdgeInfo2 for storing the signature of the
+	// lnwire.ChannelAnnouncement2 message.
+	EdgeInfo2Sig = tlv.Type(1)
```

Comment:
Can use `tlv.RecordT` now alongside the new structs. Most of these types already have records created for them as well.